vmware-event-router/hack/run_integration_tests.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+# Copyright 2019 VMware, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-2
+
+set -euo pipefail
+
+#################################
+### variables
+TIMEOUT_CMD="timeout"
+BOM_FILE="../veba-bom.json"
+JQBIN="jq -r"
+
+# colors
+RED='\033[31m'
+GREEN='\033[32m'
+RESET='\033[0m'
+
+# OpenFaaS
+OF_VERSION=$(${JQBIN} '.openfaas.version' ${BOM_FILE})                # faas-netes version to use
+OF_TIMEOUT=3m                                                         # exit if OpenFaaS is not ready within this time
+CLI_VERSION=$(${JQBIN} '.openfaas."faas-cli"["version"]' ${BOM_FILE}) # faas-cli version
+PORT_FWD="deploy/gateway 8080:8080"                                   # local/remote ports to use for port-forwarding to OpenFaaS gateway
+OKFN=of-echo                                                          # OpenFaaS function
+FAILFN=of-fail                                                        # OpenFaaS function
+FN_TIMEOUT=1m                                                         # exit if OpenFaaS functions are not ready within this time
+
+# AWS EventBridge
+AWS_SECRET=secret_aws.json # when running AWS integration tests, secret file holding AWS config at index [1]
+
+# Kubernetes (kind)
+K8S_VERSION=$(${JQBIN} '.kubernetes.version' ${BOM_FILE})
+KINDBIN="kind"
+KIND_TIMEOUT=5m # exit if kind cluster creation does not complete within this time
+KIND_WAIT=2m    # wait for control plane to show ready
+KIND_CLUSTER="veba-integration"
+# using custom image built with kind since not all versions are pushed to kindtest
+KIND_IMAGE="embano1/node:${K8S_VERSION}"
+#################################
+
+tmp_dir=$(mktemp -d -t ci-XXXXXXXXXX)
+
+cecho() {
+    local default_msg="No message passed."
+    message=${1:-$default_msg}
+    color=${2:-$RESET}
+
+    counter=$((${counter:-0} + 1))
+    echo -e ${color}"\n"\[ ${counter} \] ${message} ${RESET}
+    return
+}
+
+function cleanup() {
+    # capture previous return val, e.g. error
+    rv=$?
+    # don't run cleanup in CI env, e.g. Github Action sets CI=true
+    if [ "${CI:-no}" != true ]; then
+        cecho "Running cleanup and then exiting with code ${rv}" ${GREEN}
+        rm -rf ${tmp_dir} && rm -rf faas-cli-* && rm -rf template && ${KINDBIN} delete cluster --name ${KIND_CLUSTER}
+    fi
+    exit $rv
+}
+trap cleanup INT TERM EXIT
+
+# create kind cluster
+cecho "---> Creating Kubernetes cluster (${K8S_VERSION}) with max wait time: ${KIND_TIMEOUT}" ${GREEN}
+${TIMEOUT_CMD} ${KIND_TIMEOUT} ${KINDBIN} create cluster --name ${KIND_CLUSTER} --wait ${KIND_WAIT} --image ${KIND_IMAGE}
+
+# generate password used by OpenFaaS for basic_auth
+export OF_PASSWORD=$(head -c 12 /dev/urandom | shasum | cut -d' ' -f1)
+
+# if running as Github Actions make it available to other steps
+if [ "${CI:-no}" = true ]; then
+    echo "::set-env name=OF_PASSWORD::$OF_PASSWORD"
+fi
+
+# deploy faas-netes
+cecho "---> Deploying OpenFaaS (${OF_VERSION})" ${GREEN}
```

Comment:
Good point, it's there `helm (v3.2.4+g0ad800e)`. I would like to keep `basic_auth` on even for testing since this seems to be the default in most environments I guess.